controllers/provisioners/ekscloudformation/types.go

```diff
@@ -34,6 +36,29 @@ type AwsAuthConfigMapRolesData struct {
 	MapRoles []AwsAuthConfig `yaml:"mapRoles"`
 }
 
+func (m *AwsAuthConfigMapRolesData) AddUnique(config AwsAuthConfig) {
+	for _, existingConf := range m.MapRoles {
+		if reflect.DeepEqual(existingConf, config) {
+			return
+		}
+	}
+	if config.RoleARN == "" || config.Username == "" || len(config.Groups) == 0 {
+		return
+	}
+	m.MapRoles = append(m.MapRoles, config)
+}
+
+func (m *AwsAuthConfigMapRolesData) Remove(config AwsAuthConfig) {
+	for i := len(m.MapRoles) - 1; i >= 0; i-- {
+		role := m.MapRoles[i]
+
+		if reflect.DeepEqual(role, config) {
+			m.MapRoles = append(m.MapRoles[:i],
```

Comment:
Is this function used? I'm not following how the config is removed by `append`.

Reply:
It removes from the list by index (using append) - see `re-slicing`
But good catch - looks like it's not needed and is a remnant if previous commit, removal is now handled by not discovering deleting node groups.

I'll remove this method